cmd/dockerd/daemon.go

```diff
@@ -111,6 +110,10 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 		return err
 	}
 
+	if err := system.MkdirAll(cli.Config.ExecRoot, 0700, ""); err != nil {
```

Comment:
This looks new; where/when was this directory created in the past?

I also see some  validations steps below that may fail; is it possible to move this further down, so that we don't end up with this directory if the configuration happened to be invalid?


Finally; not sure if this directory needs special treatment (as the "daemon" root gets); also taking into account that this code is both Linux and Windows (and Windows doesn't handle permissions similar to Linux)

https://github.com/moby/moby/blob/8e2f9203065987116aec9e2d2a1d5c7039e1a5d4/daemon/daemon_unix.go#L1136-L1187

https://github.com/moby/moby/blob/6cd806aa53de41e6f04346150d08e19ec63311e0/daemon/daemon_windows.go#L472-L479

Reply:
Previously this directory was getting created as a side effect of getting a containerd client. This seemed to be an inappropriate way to create a shared directory so I just moved it here. It is still getting created with same permissions at roughly the same time though.